### PR TITLE
Refactoring to typescript

### PR DESCRIPTION
**Pull request by tomasstrejcek**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,333 +23,289 @@ const CSOBPaymentModule = class CSOBPaymentModule {
...
     })
-  };
+    if (this._verify(this._createResultMessage(result), result.signature)) {
+      if (result.resultCode.toString() === '0') {
+        return result
+      } else {
+        throw result
+      }
+    }
+    throw new Error('Init - Verification failed')
```
Comment (mzk) [inline index.js:109]: CustomError?

File: index.js [hunk 2]
```diff
@@ -20,333 +23,289 @@ const CSOBPaymentModule = class CSOBPaymentModule {
...
+      method: 'GET',
+      json: true
     })
+    const message = this._createResultMessage(result)
+
+    if (this._verify(message, result.signature)) {
+      if (result.resultCode.toString() === '0') {
+        return result
+      } else {
+        throw result
```
Comment (mzk) [inline index.js:148]: ?

File: index.js [hunk 3]
```diff
@@ -20,333 +23,289 @@ const CSOBPaymentModule = class CSOBPaymentModule {
...
   payOrder(order, close = true, options = {}) {
-    const payload = Object.assign(options, this.config.payloadTemplate);
+    const payload = Object.assign(options, this.config.payloadTemplate)
     payload['orderNo'] = order.id
-    payload['dttm'] = this._createDttm();
-    payload['description'] = order.description;
-    payload['cart'] = order.items;
-    payload['totalAmount'] = order.items.reduce((sum, item) => sum + item.amount, 0);
-    payload['closePayment'] = close;
+    payload['dttm'] = this._createDttm()
```
Comment (mzk) [inline index.js:285]: payload[
'dttm' => this._createDttm(),
...
]

**Review by mzk — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (tomasstrejcek) [reply index.js:148]: dont approve if you see something weird :)

**Pull request merged**